app/api/auth.py

```diff
@@ -345,6 +348,32 @@ def order_invoices(order_identifier):
         return ForbiddenError({'source': ''}, 'Authentication Required to access Invoice').respond()
 
 
+@ticket_blueprint.route('/events/invoices/<string:invoice_identifier>')
+@jwt_required()
+def event_invoices(invoice_identifier):
+    if current_user:
```

Comment:
There are better design methods that you can consider. For eg:

```
if not current_user:
    raise Forbidden error
```
There will be no unnecessary if-else ladder

Reply:
@poush This if-else ladder is required because at each stage, we are validating the user & the corresponding permissions. If there is any violation at any point, we throw the appropriate exception.

The example given by you is more or less the equivalent of my logic. If we use 
```
if not current_user:
    raise Forbidden error
```,
we would then have to proceed with the checks  in the else block similarly. 
```
else if current__user.is_staff and current_user.is_verified:
                ...
```

Reply:
@poush You can check the implementation given for tickets & order invoices which is akin to what I've proposed here :D

Reply:
> we would then have to proceed with the checks in the else block

No you won't 
You can assume from the next line that current_user is present, removing the need of nested conditions

Reply:
> > we would then have to proceed with the checks in the else block
> 
> No you won't
> You can assume from the next line that current_user is present, removing the need of nested conditions

Won't the nesting still be present to check for current_user.is_verified & other cases?

Reply:
```python
if current_user:
  # 1 level nesting
  if current_user.is_verified:
    # 2 level
  else:
    # 2 level
    throw
else:
  # 1 level
  throw
```
Max Level 2
Conditions 4

```python
if not current_user:
  throw

if current_user.is_verified:
  throw

# continue
```

Max level nesting: 0
Conditions 2

Reply:
@iamareebjamal @poush Understood! Will be updating this.